runtime/pattern.go

```diff
@@ -140,10 +142,11 @@ func MustPattern(p Pattern, err error) Pattern {
 	return p
 }
 
-// Match examines components if it matches to the Pattern.
-// If it matches, the function returns a mapping from field paths to their captured values.
-// If otherwise, the function returns an error.
-func (p Pattern) Match(components []string, verb string) (map[string]string, error) {
+// MatchAndEscape examines components if it matches to the Pattern. If it matches,
+// the function returns a mapping from field paths to their captured values while
+// applying the provided unescaping mode, returning an error if the URL encoding
+// is malformed. Otherwise, the function returns an error.
```

Comment:
This comment made me think we sometimes return both an error and a map.
```suggestion
// MatchAndEscape examines components to determine if it matches to the Pattern.
// If it matches, the function returns a mapping from field paths to their captured
// values while applying the provided unescaping mode, returning an error if the
// URL encoding is malformed.
```

Reply:
How about:

```
// MatchAndEscape examines components to determine if they match to a Pattern.
// MatchAndEscape will return an error if no Patterns matched or if a pattern matched
// but contained malformed escape sequences. If successful, the function returns a 
// mapping from field paths to their captured values.
```